docs/modules/i2c.md

```diff
@@ -2,170 +2,301 @@
+
+-- attempt to read chip id and compare against expected value
+function simple_check_chip( dev_address, dev_register , dev_signature )
+  i2c.start( id )
+  assert( i2c.address( id , dev_address , i2c.TRANSMITTER ) , "!!i2c device did not ACK first address operation" )
+  i2c.write( id , dev_register )
+  i2c.start( id ) -- repeated start condition
+  assert( i2c.address( id , dev_address , i2c.RECEIVER ) , "!!i2c device did not ACK second address operation" )
+  if i2c.read( id , 1):byte() == dev_signature then
+    print( "..chip is operational" )
+  else
+    print( "!!The chip does not have the expected signature" )
+  end
+   i2c.stop( id )
+end
+-- 
+print( "Chip check, should fail because device address is wrong" )
+simple_check_chip( bmpAddress+1 , bmpIdRegister , bmpChipSignature)
+
+print( "Chip check, should succeed if chip is present and functional" )
+simple_check_chip( bmpAddress , bmpIdRegister , bmpChipSignature)
+
+```
+
+#### Example using the hardware interfaces
+```lua
+sda = 23 -- pins as on Adafruit Huzzah32 silkscreen
+scl = 22
+id = i2c.HW1 -- hardware interface
+speed = i2c.SLOW
+
+-- values for Bosch Sensortech BMP180 chip
+bmpAddress = 0x77
+bmpIdRegister = 0xD0
+bmpChipSignature = 0x55
+
+-- initialize i2c software interface
+i2c.setup( id , sda , scl , speed )
+
+-- read a single byte from the chip
+function read_byte( dev_address, dev_register , callback )
+  i2c.start( id )
+  i2c.address( id , dev_address , i2c.TRANSMITTER )
+  i2c.write( id , dev_register )
+  i2c.start( id ) -- repeated start condition
+  i2c.address( id , dev_address , i2c.RECEIVER )
+  i2c.read( id , 1)
+  i2c.stop( id )
+  return i2c.transfer( id , callback )
+end
+
+-- check results returned
+function check( value , 	ack )
+  if ack then
+    if value:byte() == bmpChipSignature then
+      print( "..chip is operational" )
+    else
+      print( "!!The chip does not have the expected signature" )
+    end
+  else
+    print( "!!chip did not respond" )
+  end
+end
+--
+print( "synchronous use")
 
-Communication stops when the slave answers with NACK to the address byte. This can be avoided with parameter `ack_check_en` on `false`.
+print( "Chip check, should fail because device address is wrong" )
+check( read_byte( bmpAddress+1 , bmpIdRegister ) )
+
+print( "Chip check, should succeed if chip is present and functional" )
+check( read_byte( bmpAddress , bmpIdRegister ) )
+
+print( "asynchronous use")
+
+print( "Chip check, should fail because device address is wrong" )
+read_byte( bmpAddress+1 , bmpIdRegister, check)
+
+print( "Chip check, should succeed if chip is present and functional" )
+read_byte( bmpAddress , bmpIdRegister , check)
+```
+
+## i2c.address()
+Perform (`SW`) or enqueue (`HWx`) an I²C address operation, defining data transfer direction for the next operation (read or write).
 
 #### Syntax
-`i2c.address(id, device_addr, direction[, ack_check_en])`
+`i2c.address( id , device_addr , direction [, ack_check_en] )`
```

Comment:
For consistency reasons please don't change this (applies to the entire file). It's purely a matter of personal taste.

Reply:
@marcelstoer Thanks for your comments. Consistency is important.
I have adapted the file accordingly.
BTW there are several style guides for  Lua. Is there any recommended one to be used here ?
Regards,
jmdasnoy